platforms/software/software-diagnostics/src/main/java/org/gradle/api/plugins/ProjectReportsPlugin.java

```diff
@@ -24,11 +24,14 @@
 import org.gradle.internal.deprecation.DeprecationLogger;
 import org.gradle.util.internal.WrapUtil;
 
+import javax.annotation.Nonnull;
+
 /**
  * <p>A {@link Plugin} which adds some project visualization report tasks to a project.</p>
  *
  * @see <a href="https://docs.gradle.org/current/userguide/project_report_plugin.html">Project Report plugin reference</a>
  */
+@Nonnull
```

Comment:
Do the task names below not belong as part of `DiagnosticsTaskNames`?

Reply:
There's no need for this, per the other discussion on that interface.